src/dotnet/commands/dotnet-msbuild/MSBuildForwardingApp.cs

```diff
@@ -77,6 +79,11 @@ internal static CommandOption AddVerbosityOption(CommandLineApplication app)
             return app.Option("-v|--verbosity", LocalizableStrings.VerbosityOptionDescription, CommandOptionType.SingleValue);
         }
 
+        private static string Escape(string arg) =>
+             (arg.StartsWith("/p:RestoreSources=", StringComparison.OrdinalIgnoreCase)) ?
+                arg.Replace(";", "%3B") : // <-- this is a workaround for https://github.com/Microsoft/msbuild/issues/1622
```

Comment:
The forward slashes in the linked issue's description of the workaround are a bit misleading. The semicolon is what needs to be escaped.